### ISSUE TITLE
Ignored Exceptions

### ISSUE THREAD

**Issue by Xerus2000** [hypfvieh/dbus-java#6]
![image](https://user-images.githubusercontent.com/13354331/36133225-982b3c98-107c-11e8-806f-f19d7f7075a9.png)

This code smells badly. Exceptions are blatantly ignored, presumably because they should actually be RuntimeExceptions...

**Comment by hypfvieh** [comment]
As you might know, I'm not the original author of the code. I just took it and re-organized, cleaned various java warnings, added generics here and there and created a maven project out of it.

There are only a few things added either by myself or by some other committers (like PropertiesChangedListener).

So because of that, I expect that there are lots of ugly parts and code smells within this library.
I also really dislike the if (EXCEPTION_DEBUG) stuff, but I'm just to busy to cleanup all this mess.

But hey, if you don't like it, fix it and I will happily merge your changes.

**Comment by hypfvieh** [comment]
I just merged your changes and I also removed the 'EXCEPTION_DEBUG' flag.

It might be a good idea to review the whole code and clean this exception mess. Exceptions are thrown everywhere, logged and then being wrapped in another exception to be thrown, which in turn will be logged and maybe wrapped in the next method.. This is really bad style. 
Also catching RuntimeExceptions does not seem to be a good idea to me.
